src/main/scala/mesosphere/marathon/core/health/impl/AntiSnowballApiImpl.scala

```diff
@@ -0,0 +1,40 @@
+package mesosphere.marathon
+package core.health.impl
+
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.api.v2.json.Formats._
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.state.{AbsolutePathId, Timestamp}
+import mesosphere.marathon.core.health.{AntiSnowballApi, HealthCheckShield, HealthCheckShieldApi, HealthCheckShieldConf}
+
+import scala.collection.concurrent.TrieMap
+import scala.async.Async.{async, await}
+import akka.stream.scaladsl.{Sink, Source}
+import akka.stream.ActorMaterializer
+
+import scala.concurrent.Future
+import akka.Done
+
+import scala.concurrent.ExecutionContext
+
+/* Store info about anti-snowball status for each app
+*/
+final class AntiSnowballApiImpl()(implicit ec: ExecutionContext, mat: ActorMaterializer)
+  extends AntiSnowballApi with StrictLogging {
+
+  // maybe we'll eventually store a AntiSnowballStatus object
+  private val antiSnowballStatus = TrieMap.empty[AbsolutePathId, Boolean]
+
+  def getStatus(appId: AbsolutePathId): AntiSnowballStatus = {
+    val active = antiSnowballStatus.getOrElse(appId, false)
```

Comment:
404 is for apps that don't exist. Here it would mean that we don't have enough information to answer because we had no case of healthchecks that would normally have justified to kill the app.

After a leader election, most apps will be in that case (because most apps have passing healthchecks on all their tasks). We will enter in the code path that set the active flag only when a task starts to fail its healthcheck sufficiently enough to be a candiate for killing.

I think it is necessary to answer that antisnowball is not active when we don't have info.

Reply:
Yes I understood that after reading the first comment from you. Without the 404 thing we would have answer "false" for apps that just don't exist. It's fine as the absolute spec id is verified earlier :+1: